go-controller/pkg/ovn/egressqos.go

```diff
@@ -170,43 +170,56 @@ func (oc *DefaultNetworkController) createASForEgressQoSRule(podSelector metav1.
 func (oc *DefaultNetworkController) initEgressQoSController(
 	eqInformer egressqosinformer.EgressQoSInformer,
 	podInformer v1coreinformers.PodInformer,
-	nodeInformer v1coreinformers.NodeInformer) {
+	nodeInformer v1coreinformers.NodeInformer) error {
 	klog.Info("Setting up event handlers for EgressQoS")
 	oc.egressQoSLister = eqInformer.Lister()
 	oc.egressQoSSynced = eqInformer.Informer().HasSynced
 	oc.egressQoSQueue = workqueue.NewNamedRateLimitingQueue(
 		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
 		"egressqos",
 	)
-	eqInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+	_, err := eqInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
 		AddFunc:    oc.onEgressQoSAdd,
 		UpdateFunc: oc.onEgressQoSUpdate,
 		DeleteFunc: oc.onEgressQoSDelete,
 	}))
+	if err != nil {
+		klog.Errorf("Could not add Event Handler for eqInformer during egressqosController initialization, %v", err)
+		return err
+	}
 
 	oc.egressQoSPodLister = podInformer.Lister()
 	oc.egressQoSPodSynced = podInformer.Informer().HasSynced
 	oc.egressQoSPodQueue = workqueue.NewNamedRateLimitingQueue(
 		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
 		"egressqospods",
 	)
-	podInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+	_, err = podInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
 		AddFunc:    oc.onEgressQoSPodAdd,
 		UpdateFunc: oc.onEgressQoSPodUpdate,
 		DeleteFunc: oc.onEgressQoSPodDelete,
 	}))
+	if err != nil {
+		klog.Errorf("Could not add Event Handler for podInformer during egressqosController initialization, %v", err)
+		return err
+	}
 
 	oc.egressQoSNodeLister = nodeInformer.Lister()
 	oc.egressQoSNodeSynced = nodeInformer.Informer().HasSynced
 	oc.egressQoSNodeQueue = workqueue.NewNamedRateLimitingQueue(
 		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
 		"egressqosnodes",
 	)
-	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	_, err = nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    oc.onEgressQoSNodeAdd, // we only care about new logical switches being added
 		UpdateFunc: func(o, n interface{}) {},
 		DeleteFunc: func(obj interface{}) {},
 	})
+	if err != nil {
+		klog.Errorf("Could not add Event Handler for nodeInformer during egressqosController initialization, %v", err)
+		return err
```

Comment:
just return this as the error message and dont log it. Use %w to wrap the error.